### PR TITLE
Implement trussed-auth API

### PR DESCRIPTION
**Pull request by sosthene-nitrokey**
Since this is amongst the most security-sensible parts of the entire backend, it should be thoroughly reviewed. 

See the se050 notes and key derivation schema on the nextcloud for the approach taken.

### CODE REVIEW

**Comment by sosthene-nitrokey** [comment]
I've tested this in the firmware and it appears to work reliably, though the initial testing did lead to inconsistent state between the filesystem and the SE050 which lead to failures, this suggests that some improvements to error handling could be done and that a full device factory reset will likely be useful, at least to expose the factory reset of the SE050 some way for developers.

**Comment by tlaurion** [comment]
@sosthene-nitrokey ELI5 here would be really really appreciated for downstream projects.
What are the impacts? improvements? general status outside of reviewing the implementation? What is usable, time generation of lets say RSA3076/RSA4096 keys?

**Comment by sosthene-nitrokey** [comment]
This repo will contain an implementation of the trussed APIs through the SE050 secure element rather than the current software implementation.

This  PR adds  an implementation of the [russed-auth](https://github.com/trussed-dev/trussed-auth) APIs.
This API provides:
- Pin handling (creation and verification)
- Key derivation from PINs (in this case, rather protecting a key with a PIN).


Future PR will implement the required APIs for asymmetric crypto operations (ECC and RSA), and key wrapping.

This is currently heavily WIP and **should not** be used in production.

**Comment by tlaurion** [comment]
[Heads users having bought NK3 keys are starting to ask questions on traditional communication channels.](https://matrix.to/#/!pAlHOfxQNPXOgFGTmo:matrix.org/$9WMRPeodvSqFuyP3HOcSNl0dW0K86GZvG7o_GD6d9bo?via=matrix.org&via=nitro.chat&via=fairydust.space) I answered the best I could.

Crosslinking to https://github.com/osresearch/heads/pull/1469

**Comment by sosthene-nitrokey** [comment]
Ok, I understand. You are asking whether this will allow supporting RSA3072 generated on the device for heads support.

This work will indeed lead to that, but in another PR.
